sw/device/lib/base/mmio.h

```diff
@@ -26,6 +26,19 @@ extern "C" {
  * MMIO accesses.
  */
 
+/**
+ * 2020-06: We're transitioning to a more efficient manner of using our MMIO
+ * APIs, where the DIFs explicitly read, then modify, then write. All the
+ * `*_nonatomic_*` functions in this DIF are deprecated and will be removed
+ * eventually, leaving only the `read<N>`, `write<N>`, and `memcpy` functions.
+ *
+ * For the moment, we are not adding `__attribute__((deprecated(reason)))` using
```

Comment:
How would you feel about adding it but hiding it behind an `#if 0` for the time being?

Reply:
We need the macro defined on both sides of the if, so we'd end up with
```c
#if 0
#define MMIO_DEPRECATED …
#else 
#define MMIO_DEPRECATED
#endif
```
anyway, right?

Reply:
I knew that when I made the suggestion. Actually, maybe we should write
```
#define MMIO_DEPRECATED  // __attribute__((deprecated(...)))
```

No strong feelings either way. Feel free to resolve without making changes.